pkg/controller/llmisvc/workload_multi_node.go

```diff
@@ -531,11 +533,32 @@ func (r *LLMInferenceServiceReconciler) propagateLeaderWorkerSetMetadata(llmSvc
 			}
 			expected.Annotations[k] = v
 		}
+
+		if strings.HasPrefix(k, "k8s.v1.cni.cncf.io") {
+			ann[k] = v
+			if expected.Annotations == nil {
+				expected.Annotations = make(map[string]string, 1)
+			}
+			expected.Annotations[k] = v
+		}
 	}
 
-	expected.Spec.LeaderWorkerTemplate.WorkerTemplate.Annotations = ann
 	if expected.Spec.LeaderWorkerTemplate.LeaderTemplate != nil {
-		expected.Spec.LeaderWorkerTemplate.LeaderTemplate.Annotations = ann
+		if expected.Spec.LeaderWorkerTemplate.LeaderTemplate.Annotations == nil {
+			expected.Spec.LeaderWorkerTemplate.LeaderTemplate.Annotations = ann
+		} else {
+			for k, v := range ann {
+				expected.Spec.LeaderWorkerTemplate.LeaderTemplate.Annotations[k] = v
+			}
+		}
+	}
+
+	if expected.Spec.LeaderWorkerTemplate.WorkerTemplate.Annotations == nil {
+		expected.Spec.LeaderWorkerTemplate.WorkerTemplate.Annotations = ann
+	} else {
+		for k, v := range ann {
+			expected.Spec.LeaderWorkerTemplate.WorkerTemplate.Annotations[k] = v
+		}
```

Comment:
On the fence with this one, but maybe if map is `nil` just `make` it and "merge" in a loop with `ann`. It would eliminate the `else` blocks.

Reply:
Given strict timing I can follow up to clean this up (perhaps as part of the upstream PR I've open there)

Reply:
That is totally fine.